lib/Extension/PHPUnit/CodeTransform/GenerateTestMethods.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Phpactor\Extension\PHPUnit\CodeTransform;
+
+use Generator;
+use Phpactor\CodeBuilder\Domain\Builder\SourceCodeBuilder;
+use Phpactor\CodeBuilder\Domain\Code;
+use Phpactor\CodeBuilder\Domain\Updater;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextEdits;
+use Phpactor\WorseReflection\Core\ClassName;
+use Phpactor\WorseReflection\Core\Exception\NotFound;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionClass;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\WorseReflection\Reflector;
+use Webmozart\Assert\Assert;
+
+class GenerateTestMethods
+{
+    private const METHODS_TO_GENERATE = [
+        'setUp',
+        'tearDown',
+    ];
+
+    public function __construct(
+        private Reflector $reflector,
+        private Updater $updater,
+    ) {
+    }
+
+    /** @return Generator<string> */
+    public function getGeneratableTestMethods(SourceCode $source): Generator
+    {
+        $classes = $this->reflector->reflectClassesIn($source);
+        if (count($classes->classes()) !== 1) {
+            return;
+        }
+
+        $class = $classes->classes()->first();
+        if (!$class instanceof ReflectionClass) {
+            return;
+        }
+
+        if (!$class->isInstanceOf(ClassName::fromString('\PHPUnit\Framework\TestCase'))) {
+            return;
+        }
+
+        foreach (self::METHODS_TO_GENERATE as $methodName) {
+            if ($class->ownMembers()->methods()->byName($methodName)->count() === 0) {
+                yield $methodName;
+            }
+        }
+
+        return;
+    }
+
+    public function generateMethod(TextDocument $document, string $methodName): TextEdits
+    {
+        Assert::inArray(
+            $methodName,
+            self::METHODS_TO_GENERATE,
+            sprintf('%s can not generate "%s" with class', __CLASS__, $methodName),
+        );
+
+        $class = $this->reflector->reflectClassesIn($document)->classes()->first();
+
+        $builder = SourceCodeBuilder::create();
+        $builder->namespace($class->name()->namespace());
+        $classBuilder = $builder->class($class->name()->short());
+
+        try {
+            if ($class->methods()->has($methodName)) {
```

Comment:
this is why the method is not being added, it needs to be:

```
        if ($class->ownMembers()->methods()->has($methodName)) {
            return TextEdits::none();
        }
```

perhaps we could make the tests fail before applying the fix to make them green :+1:

Reply:
Yeah removing your fix, will break the test. (I checked.) But I have no idea how I should commit that.

Reply:
It's only a development approach (TDD) to help ensure the test is a good one. (I will often commit when the test is failing and then again when it's green)